$OpenBSD: patch-src_scrollpad_cpp,v 1.1 2015/06/19 16:23:33 dcoppa Exp $

commit aa8c47d5782664b7d9786e1195f7950e29ee9a21
Author: Andrzej Rybczak <electricityispower@gmail.com>
Date:   Sun May 17 16:30:11 2015 +0200

actions: respect regular expression config when searching in text
fields

--- src/scrollpad.cpp.orig	Sat May  2 21:16:11 2015
+++ src/scrollpad.cpp	Fri Jun 19 16:28:46 2015
@@ -27,7 +27,7 @@
 namespace {//
 
 template <typename PropT>
-bool regexSearch(NC::Buffer &buf, PropT begin, const std::string &ws, PropT end, size_t id, boost::regex::flag_type flags)
+bool regexSearch(NC::Buffer &buf, PropT begin, const std::string &ws, PropT end, boost::regex::flag_type flags, size_t id)
 {
 	try {
 		boost::regex rx(ws, flags);
@@ -274,14 +274,14 @@ void Scrollpad::reset()
 	m_beginning = 0;
 }
 
-bool Scrollpad::setProperties(Color begin, const std::string &s, Color end, size_t id, boost::regex::flag_type flags)
+bool Scrollpad::setProperties(Color begin, const std::string &s, Color end, size_t flags, size_t id)
 {
 	return regexSearch(m_buffer, begin, s, end, id, flags);
 }
 
-bool Scrollpad::setProperties(Format begin, const std::string &s, Format end, size_t id, boost::regex::flag_type flags)
+bool Scrollpad::setProperties(Format begin, const std::string &s, Format end, size_t flags, size_t id)
 {
-	return regexSearch(m_buffer, begin, s, end, id, flags);
+	return regexSearch(m_buffer, begin, s, end, flags, id);
 }
 
 void Scrollpad::removeProperties(size_t id)
